### PR TITLE
feat: restore web backup

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Support web backups with `desktop_wbu` extension for testing purposes. Currently it supports only text messages
Also added small fix in `ExternalContactSearchResultItem` to log unhandled connection state

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1551/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt [hunk 1]
```diff
@@ -142,7 +142,7 @@ fun ExternalContactSearchResultItem(
         },
         actions = {
             when (connectionState) {
-                ConnectionState.NOT_CONNECTED ->
+                ConnectionState.NOT_CONNECTED, ConnectionState.CANCELLED ->
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt:145]: Is this change on the scope of this PR?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt [hunk 2]
```diff
@@ -151,7 +151,7 @@ fun ExternalContactSearchResultItem(
                 ConnectionState.BLOCKED -> {
                 }
                 else -> {
-                    appLogger.e("Unknown ConnectionStatus in InternalContactSearchResultItem")
+                    appLogger.e("Unknown ConnectionStatus in InternalContactSearchResultItem $connectionState")
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt]: As `connectionState` is an enum class known in runtime, I would rather add the missing `ACCEPTED` and `MISSING_LEGALHOLD_CONSENT` cases in order to be exhaustive and not have this unnecessary else.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1551/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5991860)(5991860)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5991860)(5991860)-staging-compat.apk)

**Review by gongracr — APPROVED** [review]
Just a tiny non-blocking comment. But looking 🍇 otherwise 🙌

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt:145]: Maybe this change is related to your other ticket AR-2086 and got here by accident? 🤔

Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt:145]: Yeah, I saw some error messages and by mistake added it here, can it stay?

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/InternalContactSearchResultItem.kt]: Good point 👍

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1551/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5992270)(5992270)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5992270)(5992270)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1551/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5993750)(5993750)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5993750)(5993750)-staging-compat.apk)

**Pull request merged**